agent-inject/agent/config_test.go

```diff
@@ -142,6 +142,96 @@ func TestNewConfig(t *testing.T) {
 	}
 }
 
+func TestFilePathAndName(t *testing.T) {
+
+	tests := []struct {
+		name        string
+		annotations map[string]string
+		destination string
+	}{
+		{
+			"just secret",
+			map[string]string{
+				"vault.hashicorp.com/agent-inject-secret-foo": "db/creds/foo",
+			},
+			secretVolumePath + "/foo",
+		},
+		{
+			"with relative file path",
+			map[string]string{
+				"vault.hashicorp.com/agent-inject-secret-foo": "db/creds/foo",
+				"vault.hashicorp.com/agent-inject-file-foo":   "nested/foofile",
+			},
+			secretVolumePath + "/nested/foofile",
+		},
+		{
+			"with absolute file path",
+			map[string]string{
+				"vault.hashicorp.com/agent-inject-secret-foo": "db/creds/foo",
+				"vault.hashicorp.com/agent-inject-file-foo":   "/special/volume/foofile",
+			},
+			secretVolumePath + "/special/volume/foofile",
+		},
+		{
+			"with volume mount set, file name",
+			map[string]string{
+				"vault.hashicorp.com/agent-inject-secret-foo": "db/creds/foo",
+				"vault.hashicorp.com/agent-inject-file-foo":   "foofile_name_is_very_very_very_long",
+				AnnotationVaultSecretVolumePath:               "/new/mount/path",
+			},
+			"/new/mount/path/foofile_name_is_very_very_very_long",
+		},
+		{
+			"with volume mount set, absolute file path",
+			map[string]string{
+				"vault.hashicorp.com/agent-inject-secret-foo": "db/creds/foo",
+				"vault.hashicorp.com/agent-inject-file-foo":   "/special/foofile",
+				AnnotationVaultSecretVolumePath:               "/new/mount/path",
+			},
+			"/new/mount/path/special/foofile",
+		},
+		{
+			"with volume mount set, relative file path",
+			map[string]string{
+				"vault.hashicorp.com/agent-inject-secret-foo": "db/creds/foo",
+				"vault.hashicorp.com/agent-inject-file-foo":   "nested/foofile",
+				AnnotationVaultSecretVolumePath + "-foo":      "/new/mount/path",
+			},
+			"/new/mount/path/nested/foofile",
+		},
```

Comment:
What is the effective difference between these two scenarios? The `inject-file-foo` value is different by the leading `/` but the both end up in the same file structure:

```
/new/mount/path/special/foofile
```
vs
```
/new/mount/path/nested/foofile
```

Reply:
The leading `/` is the main difference, just making sure both path styles work when a custom mount path is specified. Tried to make the test descriptions a little more clearer in b96d6b4 if that helps.